### PR TITLE
#156874253 Decode token and return User Details

### PR DESCRIPTION
**Pull request by JoyLubega**
### What does this PR do?
This PR decode the token and returns user's Name and email

### How should this be manually tested?
N/A

### What are the relevant Pivotal Tracker board stories?
#156874253
### Screenshots (if appropriate)
screen shot 2018-05-17 at 01 40 16
screen shot 2018-05-17 at 00 37 53

### CODE REVIEW

**Comment by PatrickCmd** [comment]
I am curious, why have you included pull request template here.

**Comment by JoyLubega** [comment]
Am still working on making sure that circleci passes, Because my tests run fine on the local machine

**Comment by JoyLubega** [comment]
##Screenshots
screen shot 2018-05-23 at 23 38 56

**Comment by JoyLubega** [comment]
##Screenshots
screen shot 2018-05-23 at 23 38 56

**Comment by JoyLubega** [comment]
Some help needed, to make circleCI pass I had to hard code the secret key in my work. CircleCI couldn't pick it from the env.

**Comment by PatrickCmd** [comment]
@JoyLubega see @dkam26 to add that environment variable on CircleCI for the mrm project.

**Comment by JoyLubega** [comment]
FInally, CircleCI is Passing

**Comment by JoyLubega** [comment]
@PatrickCmd I have removed the print statements

File: admin_schema.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import graphene
```
Comment (shemogumbe) [inline admin_schema.py]: User model not necesary for the sake of decoding a token received from front end as part of response headers

File: api/user/models.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy import (
```
Comment (shemogumbe) [inline api/user/models.py]: For the sake of the task 'decoding token'this model is not necesary

File: api/user/models.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    Column, String, Integer, ForeignKey, func, 
+    DateTime, create_engine)
+from sqlalchemy.orm import relationship
+
+from helpers.database import Base
+from utilities.utility import Utility
+
+
+class User(Base, Utility):
```
Comment (shemogumbe) [inline api/user/models.py]: Not necesary for the sake of decoding token

File: api/user/schema.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    def resolve_users(self, info):
+        query = User.get_query(info)
+        return query.all()
+
+    def resolve_user(self, info, email):
+       query = User.get_query(info)
+       return query.filter(UserModel.email == email).first()
+
+
+class Mutation(graphene.ObjectType):
```
Comment (shemogumbe) [inline api/user/schema.py]: Not necesary for purposes of the task decoding token

**Comment by JoyLubega** [comment]
@shemogumbe , Help me elaborate on what is unnecessary

File: helpers/auth/decode_token.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+class Authentication():
+  """ Authenicate token 
+    :methods
+        verify
+        decode_token
+        is_admin
+  """
+
+  def verify(self, token):
```
Comment (shemogumbe) [inline helpers/auth/decode_token.py]: Makes sense, issues is, where does the token come from ,  does not seem like it comes from the request headers?

File: helpers/auth/decode_token.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    try:
+      payload = jwt.decode(auth_token, SECRET_KEY)
+      return payload
+    except jwt.ExpiredSignatureError:
+      return jsonify({ 'message':'Signature expired. Please log in again.'}), 401
+
+    except jwt.InvalidTokenError:
+      return jsonify({ 'message': 'Invalid token. Please Provide a valid token!'}), 401
+
+  def is_admin(self, token):
```
Comment (shemogumbe) [inline helpers/auth/decode_token.py]: This is not necesary for now, we are not yet at the stage of permissions yet

File: helpers/auth/decode_token.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        """
+
+        if token:
+            return self.decode_token(token)
+        return jsonify(
+            {'message': 'This endpoint requires you to be authenticated.'}), 401
+
+
+    def resolve_token(self, info):
+        print(info)
```
Comment (PatrickCmd) [inline helpers/auth/decode_token.py]: print statements should not be pushed, please remove them.

File: helpers/auth/decode_token.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    def decode_token(self, auth_token):
+        """ Decodes the auth token
+        :param auth_token:
+        :return: integer|string
+        """
+        SECRET_KEY = os.getenv('SECRET_KEY')
+        try:
+            payload = jwt.decode(auth_token, verify=False)
+            print(payload)
```
Comment (PatrickCmd) [inline helpers/auth/decode_token.py]: Same as above.

**Pull request closed without merging**